DigitalLearningSolutions.Web/ViewModels/Common/BaseSearchablePageViewModel.cs

```diff
@@ -22,29 +23,36 @@ public abstract class BaseSearchablePageViewModel
             string? searchString,
             string sortBy,
             string sortDirection,
+            string? filterString,
             int page,
             int itemsPerPage = 10
         )
         {
             SortBy = sortBy;
             SortDirection = sortDirection;
             SearchString = searchString;
+            FilterString = filterString;
             Page = page;
+            Filters = new List<((string, string), IEnumerable<(string, string)>)>();
             this.itemsPerPage = itemsPerPage;
         }
-        
+
         public string SortDirection { get; set; }
-        
+
         public string SortBy { get; set; }
-        
+
         public int Page { get; protected set; }
 
         public int TotalPages { get; protected set; }
 
-        public IEnumerable<SelectListItem> SortBySelectListItems => SelectListHelper.MapOptionsToSelectListItems(SortOptions);
+        public IEnumerable<SelectListItem> SortBySelectListItems =>
+            SelectListHelper.MapOptionsToSelectListItems(SortOptions);
 
         public abstract IEnumerable<(string, string)> SortOptions { get; }
 
+        public IEnumerable<((string FilterProperty, string FilterName) Filter,
+            IEnumerable<(string DisplayText, string Filter)> FilterOptions)> Filters { get; set; }

```

Comment:
It might be possible to reduce this down to `IEnumerable<(string, IEnumerable<(string, string)>)>`, The FilterProperty currently isn't really used anywhere important, but I thought it might be needed for the javascript version.

The inner IEnumerable has to stay (whether in a tuple or new model) since it needs to maintain compatibility with the FilterOptions that are also used for the actual filtering. I don't think a model is better than the smaller IEnumerable mentioned above, but it might be better than the currently implemented one. Might be another one to leave for when I'm implementing the JS version

Reply:
I'm of the opinion that making a new model would be good here, even if it ends up getting reduced down to the slightly simpler version, just so that this Filters property (and the place where it is used in the CurrentFiltersViewComponent) are wrapped in a slightly nicer package. But I'm happy for it to be left until the JS filtering is being done to see if its worth it.

E.g. This line would become something like:
 public IEnumerable<FilterDetails> Filters { get; set; }

With a new FilterDetails model being defined with properties:
(string FilterProperty, string FilterName) Filter,
 IEnumerable<(string DisplayText, string Filter)> FilterOptions

At that point you might be able to take FilterProperty and FilterName out of the Filter tuple and put them in their own properties, but I'm not sure how drastically that would affect the logic.

Reply:
Seems to work OK changed to a model. Hopefully it won't cause any issues later.